flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -791,6 +793,9 @@ public StreamStatusMaintainer getStreamStatusMaintainer() {
 		CompletableFuture<Boolean> result = new CompletableFuture<>();
 		mainMailboxExecutor.execute(
 				() -> {
+					latestAsyncCheckpointStartDelayNanos = 1_000_000 * Math.max(
```

Comment:
During the offline discussion, it seemed the right thing to update the metric here.
But now I'm wondering,  isn't it too late to update it here?

If the 1st checkpoint is delayed for 1 hour, then the metric will report zero-delay for this one hour, right? Is it an issue?

I see the following way to solve this:
1. introduce `pendingCheckpointTimestamp`
1. upon receiving RPC call, update `pendingCheckpointTimestamp` it from cp metadata and reset `latestDelay`
1. upon starting checkpoint, do the opposite: reset `pendingCheckpointTimestamp` and update `latestDelay`
1. in `getAsyncCheckpointStartDelayNanos` report  `now() - pendingCheckpointTimestamp` or `latestDelay` depending on what is set

(probably need to adjust for multiple concurrent checkpoints)

Reply:
> If the 1st checkpoint is delayed for 1 hour, then the metric will report zero-delay for this one hour, right? Is it an issue?

Yes you are right, there will be "blind spot" period while the first checkpoint is still on-going. I think this is fine:

1. it's the same behaviour as for the network tasks
2. I think the prime use cases is to look at the statistics of finished/completed checkpoints. 

So I wouldn't complicate the code because of that.

Reply:
Makes sense.